Add APM Opentelemetry Python, Ruby, and Node docs info (#8093)
DataDog/documentation: content/en/tracing/custom_instrumentation/ruby.md

| `OpenTracing::FORMAT_RACK`     | Yes        | Because of the loss of resolution in the Rack format, note that baggage items with names containing either upper case characters or `-` are be converted to lower case and `_` in a round-trip, respectively. Datadog recommends avoiding these characters or accommodating accordingly on the receiving end. |
| `OpenTracing::FORMAT_BINARY`   | No         |                                                                                                                                                                                                                                                                                                               |

## OpenTelemetry

OpenTelemetry support is available by using the `opentelemetry-exporters-datadog` gem to export traces from OpenTelemetry to Datadog.

> 

Reviewer: <div class="alert alert-warning">
This feature is currently in beta. <a href="https://docs.datadoghq.com/help/">Reach out to support</a> if it doesn't work as you expect.
</div>

Author: 👍 updated

### Installation

- If you use [bundler][11], include the following in your `Gemfile`:

```
gem 'opentelemetry-exporters-datadog'
gem 'opentelemetry-api', '~> 0.5'
gem 'opentelemetry-sdk', '~> 0.5'
```

- Or install the gems directly using:

```
gem install opentelemetry-api
gem install opentelemetry-sdk
gem install opentelemetry-exporters-datadog
```

### Usage

> Install the datadog processor and exporter in your application and configure the options. Then use the OpenTelemetry interfaces to produces traces and other information:

Author: 👍 updated

```ruby
require 'opentelemetry/sdk'
require 'opentelemetry-exporters-datadog'

> # Configure the sdk with custom export

Author: 👍 updated

OpenTelemetry::SDK.configure do |c|
  c.add_span_processor(
    OpenTelemetry::Exporters::Datadog::DatadogSpanProcessor.new(
      OpenTelemetry::Exporters::Datadog::Exporter.new(
        service_name: 'my_service', agent_url: 'http://localhost:8126'
      )
    )
  )
end

> # For propagation of datadog specific distibuted tracing headers,

Author: 👍 updated

> # set http propagation to the Composite Propagator

Author: 👍 updated

OpenTelemetry::Exporters::Datadog::Propagator.auto_configure

> # To start a trace you need to get a Tracer from the TracerProvider

Author: 👍 updated

tracer = OpenTelemetry.tracer_provider.tracer('my_app_or_gem', '0.1.0')

# create a span
tracer.in_span('foo') do |span|
  # set an attribute
  span.set_attribute('platform', 'osx')
  # add an event
  span.add_event(name: 'event in bar')
  # create bar as child of foo
  tracer.in_span('bar') do |child_span|
    # inspect the span
    pp child_span
  end
end
```

> ### Configuration Options - Datadog Agent URL

Reviewer 2: ```suggestion
### Configuration Options

The Datadog Agent URL and span tag values can be configured if necessary or desired based upon your environment and Agent location.

#### Datadog Agent URL
```

Author: 👍 updated

> By default the OpenTelemetry Datadog Exporter transmits traces to `http://localhost:8126`. You can configure the application to send traces to a diffent URL using the following environmennt variables:

Reviewer: ```suggestion
By default, the OpenTelemetry Datadog Exporter sends traces to `http://localhost:8126`. Send traces to a different URL by configuring the following environment variables:
```

Author: 👍 updated

>  - `DD_TRACE_AGENT_URL`: The `<host>:<port:` where you Datadog Agent is listening for traces. (e.g. `http://agent-host:8126`)

Reviewer: ```suggestion
 - `DD_TRACE_AGENT_URL`: The `<host>:<port>` where Datadog Agent is listening for traces, for example, `http://agent-host:8126`.
```

Author: 👍 updated

These values can also be overridden at the trace exporter level:

```ruby
# Configure the sdk with custom export
OpenTelemetry::SDK.configure do |c|
  c.add_span_processor(
    OpenTelemetry::Exporters::Datadog::DatadogSpanProcessor.new(
      OpenTelemetry::Exporters::Datadog::Exporter.new(
        service_name: 'my_service',
        agent_url: 'http://dd-agent:8126',
      )
    )
  )
end
```

> ### Configuration Options - Tagging

Reviewer 2: ```suggestion
#### Tagging
```

Author: 👍 updated

> You can configure the application to automatically tag your Datadog exported traces, using the following environment variables:

Reviewer: ```suggestion
Configure the application to automatically tag your Datadog exported traces by setting the following environment variables:
```

Author: 👍 updated

>  - `DD_ENV`: Your application environment (e.g. `production`, `staging`, etc.)

Reviewer: ```suggestion
 - `DD_ENV`: Your application environment, for example `production`, `staging`.
```

Author: 👍 updated

>  - `DD_SERVICE`: Your application's default service name (e.g. `billing-api`)

Reviewer: ```suggestion
 - `DD_SERVICE`: Your application's default service name, for example, `billing-api`.
```

Author: 👍 updated

>  - `DD_VERSION`: Your application version (e.g. `2.5`, `202003181415`, `1.3-alpha`, etc.)

Reviewer: ```suggestion
 - `DD_VERSION`: Your application version, for example, `2.5`, `202003181415`, or `1.3-alpha`.
```

Author: 👍 updated

>  - `DD_TAGS`: Custom tags in value pairs separated by `,` (e.g. `layer:api,team:intake`)

Reviewer: ```suggestion
 - `DD_TAGS`: Custom tags in value pairs, separated by commas, for example, `layer:api,team:intake`.
```

Author: 👍 updated

>     - If `DD_ENV`, `DD_SERVICE` or `DD_VERSION` are set, it will override any respective `env`/`service`/`version` tag defined in `DD_TAGS`.

Reviewer: ```suggestion
If `DD_ENV`, `DD_SERVICE`, or `DD_VERSION` is set, it will override any corresponding `env`, `service`, or `version` tag defined in `DD_TAGS`.
```

Author: 👍 updated

>     - If `DD_ENV`, `DD_SERVICE` or `DD_VERSION` are NOT set, tags defined in `DD_TAGS` will be used to populate `env`/`service`/`version` respectively.

Reviewer: ```suggestion
and `DD_VERSION` are _not_ set, you can configure environment, service, and version by using corresponding tags in `DD_TAGS`.
```

Reviewer: (same comment as in nodejs page)

Author: gotcha, yup same thing applies, this is considerably easier to read, ty will update

Author: yup, makes sense, updated

> These values can also be overridden at the trace exporter level:

Reviewer: ```suggestion
You can override these values at the trace exporter level:
```

Author: 👍 updated

```ruby

> # Configure the sdk with custom export

Reviewer: ```suggestion
# Configure the SDK with custom export
```

Author: 👍 updated

OpenTelemetry::SDK.configure do |c|
  c.add_span_processor(
    OpenTelemetry::Exporters::Datadog::DatadogSpanProcessor.new(
      OpenTelemetry::Exporters::Datadog::Exporter.new(
        service_name: 'my_service',
        agent_url: 'http://localhost:8126',
        env: 'prod',
        version: '1.5-alpha',
        tags: 'team:ops,region:west'
      )
    )
  )
end
```

> This enables you to set this value on a per application basis, so you can have for example several applications reporting for different environments on the same host.

Reviewer: ```suggestion
Tag values can also be overridden at the trace exporter level. This lets you set values on a per-application basis, so you can have multiple applications reporting for different environments on the same host.
```

Reviewer: as in the nodejs page, move this sentence above the code that shows how to do it. People want to know why they're doing something before they read how to do it.

Author: appreciate the feedback that makes a lot of sense, updated here

> Tags can also be set directly on individual spans, which will supersede any conflicting tags defined at the application level.

Reviewer: ```suggestion
Tags that are set directly on individual spans supersede conflicting tags defined at the application level.
```

Author: 👍 updated
